CseEightselectBasic/Services/Export/Export.php

```
@@ -0,0 +1,105 @@
+    $this->pluginConfig = $container->get('cse_eightselect_basic.dependencies.provider')->getPluginConfig();
+  }
+
+  private function getFormatParam($request) {
+    return $request->getParam('format');
+  }
+
+  private function isTidAndFidInRequest ($request) {
+    $tenantId = $request->getHeader('8select-com-tid');
+    $feedId = $request->getHeader('8select-com-fid');
+    
+    return $tenantId != null && $feedId != null;
+  }
+
+  private function isValidFormatSpecified ($request) {
+    $format = self::getFormatParam($request);
+    
+    if ($format == null) {
+      return false;
+    }
+    
+    $validFormats = ['product_feed', 'property_feed', 'status_feed'];
+    $isFormatValid = in_array($format, $validFormats, true);
+    
+    return $isFormatValid;
+  }
+
+  private function isTidAndFidValid ($request) {
+    $tenantIdMatches = $request->getHeader('8select-com-tid') === $this->pluginConfig['CseEightselectBasicApiId'];
+    $feedIdMatches = $request->getHeader('8select-com-fid') === $this->pluginConfig['CseEightselectBasicFeedId'];
+    return $tenantIdMatches && $feedIdMatches;
+  }
+
+  public function getOffsetAndLimit ($request) {
+    //@todo maybe use symfony
+    $MAX_LIMIT = 100;
+    $limit = $request->getParam('limit', 50);
+    $offset = $request->getParam('offset', 0);
+
+    if(!is_numeric($limit) || !is_numeric($offset)) {
+      return false;
+    }
+
+    $limit = intval($limit);
+    $offset = intval($offset);
+  
+    if($limit > $MAX_LIMIT) {
+      return false;
+    }
+
+    return ['limit' => $limit, 'offset' => $offset];
+  }
+
+  public function validateExportRequest($response, $request) {
+    $isTidAndFidInRequest = self::isTidAndFidInRequest($request);
+    $isValidFormatSpecified = self::isValidFormatSpecified($request);
+
+    if (!$isTidAndFidInRequest || !$isValidFormatSpecified) {
+      $response->setHttpResponseCode(404);
+      return false;
```

do we want to return an error message telling the requester that `tid` and `fid` or the given `format` aren't provided?

---

no we explicitly want to hide the api in those cases